plugins/gatsby-source-jenkinsplugins/gatsby-node.js

```diff
@@ -0,0 +1,224 @@
+/* eslint-disable no-console */
+const axios = require('axios');
+const crypto = require('crypto');
+const path = require('path');
+const fs = require('fs');
+
+/*
+plugins: `
+  firstRelease: Date
+  buildDate: Date
+  categories: {
+    id: String!
+  }
+  dependencies: {
+    name: String
+    title: String
+    optional: Boolean
+    version: String
+    implied: Boolean
+  }
+  maintainers: {
+    id: String
+    name: String
+    email: String
+  }
+  excerpt: String
+  gav: String
+  labels: {
+    id: String!
+  }
+  name: String
+  previousTimestamp: Date
+  previousVersion: String
+  releaseTimestamp: Date
+  requiredCore: String
+  scm: {
+    issues: String
+    link: String
+    inLatestRelease: String
+    sinceLatestRelease: String
+    pullRequest: String
+  }
+  sha1: String
+  // stats:
+  title: String
+  url: String
+  version: String
+  // securityWarnings:
+  wiki: {
+    content: String
+    url: String
+  }
+`
+*/
+
+const requestGET = ({ url, reporter }) => {
+  const activity = reporter.activityTimer(`Fetching '${url}'`);
+  activity.start();
+
+  return axios
+    .get(url)
+    .then((results) => {
+      activity.end();
+      
+      if (results.status !== 200) {
+        throw results.data;
+      }
+      return results.data;
+    });
+};
+
+const getPluginSiteVersion = () => {
+  const file = path.join(__dirname, '../../GIT_COMMIT');
+  try {
+    return fs.existsSync(file) ? fs.readFileSync(file, 'utf-8').substring(0, 7) : 'TBD';
+  } catch (err) {
+    console.error(`Problem accessing ${file}`, err);
+  }
+};
+
+const fetchPluginData = async ({ createNode, reporter }) => {
+  const sectionActivity = reporter.activityTimer('fetch plugins info');
+  sectionActivity.start();
+  let page = 1;
+  const promises = [];
+  while (1) {
```

Comment:
should this be a do while loop instead? there's a condition that breaks out of this loop

Reply:
yes... but i'd have to move some variables around, so lemme think about it.